pkg/controller/grouppermission/grouppermission_controller.go

```diff
@@ -90,30 +91,21 @@ func (r *ReconcileGroupPermission) Reconcile(request reconcile.Request) (reconci
 	clusterRoleList := &v1.ClusterRoleList{}
 	err = r.client.Get(context.TODO(), request.NamespacedName, clusterRoleList)
 	if err != nil {
-		if errors.IsNotFound(err) {
-			return reconcile.Result{}, nil
-		}
 		// error reading the object - requeue the request
 		return reconcile.Result{}, err
 	}
 
-	// get permission type, empty at first
-	permission := &managedv1alpha1.Permission{}
-
-	// get clusterRoleName from permission type
-	clusterRoleName := permission.ClusterRoleName
-
 	// if crClusterRoleNameList returns true with values in there
-	allClusterRoleNameInCluster, crClusterRoleNameList := checkClusterRoleName(instance, clusterRoleList)
-	if allClusterRoleNameInCluster == false {
+	allClusterRolesOnCluster, crClusterRoleNameList := checkClusterRoleNames(instance, clusterRoleList)
+	if !allClusterRolesOnCluster {
 		for _, crClusterRoleName := range crClusterRoleNameList {
 			// get the conditions array
 			groupPermissionConditions := instance.Status.Conditions
 
 			// make new condition
 			newCondition := managedv1alpha1.Condition{
```

Comment:
Since we are updating the condition with the same logic in a few different places in the reconcile loop, can this logic be moved to a function?

Reply:
this logic for updating condition only exists in 2 places (clusterRole and clusterRolebinding). we have 2 different helper function already. not sure if refactoring it more makes sense?

Reply:
nvm misunderstood